src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CancellationTokenParametersMustComeLast.cs

```diff
@@ -61,6 +63,20 @@ public override void Initialize(AnalysisContext context)
                         last--;
                     }
 
+                    // Ignore parameters that have any of these attributes.
+                    // C# reserved attributes: https://docs.microsoft.com/dotnet/csharp/language-reference/attributes/caller-information
+                    var builder = ImmutableHashSet.CreateBuilder<INamedTypeSymbol>();
+                    builder.AddIfNotNull(compilationContext.Compilation.GetOrCreateTypeByMetadataName(typeof(CallerFilePathAttribute).FullName));
+                    builder.AddIfNotNull(compilationContext.Compilation.GetOrCreateTypeByMetadataName(typeof(CallerLineNumberAttribute).FullName));
+                    builder.AddIfNotNull(compilationContext.Compilation.GetOrCreateTypeByMetadataName(typeof(CallerMemberNameAttribute).FullName));
```

Comment:
Fetching well know types must be done in the compilation start action. Please move this outside the symbol action callback.

Reply:
Additionally, please define [WellKnownTypeNames](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/WellKnownTypeNames.cs) for each of these attribute types instead of `typeof(CallerFilePathAttribute).FullName`

Reply:
Oh I totally missed those!